### ISSUE TITLE
Unsound unsafe in netlink-packet

### ISSUE THREAD

**Issue by little-dude** [little-dude/netlink#16]
Clippy does not like the way we implemented `netlink_packet::audit::status::StatusMessage::{to_bytes,from_bytes}`:

```rust
    fn from_bytes(buf: &[u8]) -> Result<Self, DecodeError> {
        if buf.len() != size_of::<Self>() {
            return Err(format!(
                "invalid status message, expected length {}, got {}",
                size_of::<Self>(),
                buf.len()
            )
            .into());
        }
        Ok(unsafe { ptr::read(buf.as_ptr() as *const Self) })
    }

    fn to_bytes(&self, buf: &mut [u8]) {
        unsafe { ptr::write(buf.as_mut_ptr() as *mut Self, self.clone()) }
    }
```

I'm not familiar enough with unsafe to tell whether this is really UB (we do check the buffer length before casting, normally), so I think it's best to just avoid `unsafe` and rewrite this withouth it... We did the same for `NativeNla` though and there are lots of attributes that implement it, so it may add a significant amount of code... Maybe a macro could help here.

**Comment by little-dude** [comment]
At the very least, `to_bytes` should be marked as unsafe because it's UB for sure if we don't check the length of `buf` before calling the method.

**Comment by little-dude** [comment]
fixed in #17

**Comment by Shnatsel** [comment]
Looks like `to_bytes()` could be used to trigger an out-of-bounds write from safe code. This can leard to an exploitable security vulnerability. Please file a security advisory at https://github.com/RustSec/advisory-db so that users of this crate can check if they're using a vulnerable version and upgrade.

Please also consider yanking affected versions from crates.io.
